lib/Graph/Graph.cpp

```diff
@@ -1758,21 +1781,22 @@ static Constant *quantizeDataForFusedRowwiseQuantizedSparseLengthsWeightedSum(
 FusedRowwiseQuantizedSparseLengthsWeightedSumNode *
 Function::createFusedRowwiseQuantizedSparseLengthsWeightedSum(
     llvm::StringRef name, Tensor &data, NodeValue weights, NodeValue indices,
-    NodeValue lengths, ElemKind precision, bool useFP16Accumulation) {
+    NodeValue lengths, ElemKind precision, bool useFP16Accumulation,
+    ElemKind fusedElemKind) {
   Constant *rwqData =
-      quantizeDataForFusedRowwiseQuantizedSparseLengthsWeightedSum(this, data,
-                                                                   precision);
+      quantizeDataForFusedRowwiseQuantizedSparseLengthsWeightedSum(
+          this, data, fusedElemKind);
   return createFusedRowwiseQuantizedSparseLengthsWeightedSum(
       name, rwqData, weights, indices, lengths, precision, useFP16Accumulation);
 }
 
 FusedRowwiseQuantizedSparseLengthsSumNode *
 Function::createFusedRowwiseQuantizedSparseLengthsSum(
     llvm::StringRef name, Tensor &data, NodeValue indices, NodeValue lengths,
-    ElemKind precision, bool useFP16Accumulation) {
+    ElemKind precision, bool useFP16Accumulation, ElemKind fusedElemKind) {
```

Comment:
since now there is fusedElemKind, do we still need precision?  I mean, for 4bit, it must be float16Ty.

Reply:
Good point @beicy -- I added a new commit where I went through and cleaned up all the APIs that don't need precision. Now we will pass the fused ElemKind and then infer the precision from the equivalent ElemKind used by scale/offset.